sentinelsat/products.py

```diff
@@ -86,28 +86,25 @@ def nodefilter(node_info: dict) -> bool:
     """
 
     def _path_to_url(self, product_info, path, urltype=None):
-        data = dict(id=product_info["id"], title=product_info["title"])
-        data["api_url"] = self.api_url
-        data["path"] = "/".join(["Nodes('{}')".format(item) for item in path.split("/")])
+        id = product_info["id"]
+        title = product_info["title"]
+        path = "/".join(["Nodes('{}')".format(item) for item in path.split("/")])
         if urltype == "value":
-            data["urltype"] = "/$value"
+            urltype = "/$value"
         elif urltype == "json":
-            data["urltype"] = "?$format=json"
+            urltype = "?$format=json"
         elif urltype == "full":
-            data["urltype"] = "?$format=json&$expand=Attributes"
+            urltype = "?$format=json&$expand=Attributes"
         elif urltype is None:
-            data["urltype"] = ""
-        else:
-            data["urltype"] = urltype
-        return "{api_url}odata/v1/Products('{id}')/Nodes('{title}.SAFE')/{path}{urltype}".format(
-            **data
-        )
+            urltype = ""
+        # else: pass urltype as is
+        return f"{self.api_url}odata/v1/Products('{id}')/Nodes('{title}.SAFE')/{path}{urltype}"
```

Comment:
+1, but you need to drop support Python 3.5 to use it. In ``setap.py`` 3.5 is still listed.